middlewares/team.middleware.js

```diff
@@ -54,7 +54,42 @@ async function ensureUniqueHackerId(req, res, next) {
 
 /**
  * @async
- * @function ensureSpance
+ * @function createTeam
+ * @param {{body: {teamDetails: {_id: ObjectId, name: string, members: ObjectId[], devpostURL?: string, projectName: string}}}} req
+ * @param {*} res
+ * @description create a team from information in req.body.teamDetails.
+ */
+async function createTeam(req, res, next) {
+    const teamDetails = req.body.teamDetails;
+
+    const team = await Services.Team.createTeam(teamDetails);
+
+    if (!team) {
+        return res.status(500).json({
+            message: Constants.Error.TEAM_CREATE_500_MESSAGE,
+            data: {}
+        });
+    }
+
+    for (const hackerId of teamDetails.members) {
+        const hacker = await Services.Hacker.updateOne(hackerId, {
+            teamId: team._id
+        });
+
+        if (!hacker) {
```

Comment:
Do we need to perform this check?

Reply:
I think it's a good check in case some error happens with db, or something to cause the update to fail.